datastream-server/src/main/java/com/linkedin/datastream/server/Coordinator.java

```diff
@@ -583,6 +584,8 @@ private void handleAssignmentChange(boolean isDatastreamUpdate) throws TimeoutEx
       }
     });
 
+    int totalTasks = currentAssignment.values().stream().mapToInt(List::size).sum();
+
```

Comment:
call `List<String> assignment = _adapter.getInstanceAssignment(_adapter.getInstanceName());` may become an expensive operation putting a direct load on the zookeeper server, in case of unlimited retries. I have a feeling, you will need the max retry logic as well, in case this issue happens in the production and makes the zookeeper server really busy with the `getChildren` call.

Reply:
Done